java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -1395,4 +1395,11 @@ public int hashCode(int index, ArrowBufHasher hasher) {
   public <OUT, IN> OUT accept(VectorVisitor<OUT, IN> visitor, IN value) {
     return visitor.visit(this, value);
   }
+
+  /**
+   * Gets the ending offset of a record, given its index.
+   */
+  protected final int getEndOffset(int index) {
```

Comment:
why protected?  I think the intent is to make this public?

Reply:
sorry, because the getStartOffset is protected and I didn't notice that. Now I have made getStartOffset/getEndOffset public.